--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -126,7 +126,7 @@
 	installed_image_path="usr/bin/linux-$version"
 	;;
 parisc|mips|powerpc|riscv*)
-	installed_image_path="boot/vmlinux-$version"
+	installed_image_path="boot/Image"
 	;;
 *)
 	installed_image_path="boot/vmlinuz-$version"
@@ -149,13 +149,14 @@
 else
 	cp System.map "$tmpdir/boot/System.map-$version"
 	cp $KCONFIG_CONFIG "$tmpdir/boot/config-$version"
+	cp ${srctree}/arch/$SRCARCH/boot/Image $tmpdir/boot/Image
 fi
 cp "$($MAKE -s -f $srctree/Makefile image_name)" "$tmpdir/$installed_image_path"
 
 if is_enabled CONFIG_OF_EARLY_FLATTREE; then
 	# Only some architectures with OF support have this target
 	if [ -d "${srctree}/arch/$SRCARCH/boot/dts" ]; then
-		$MAKE -f $srctree/Makefile INSTALL_DTBS_PATH="$tmpdir/boot/dtbs/$packagename" dtbs_install
+		$MAKE -f $srctree/Makefile INSTALL_DTBS_PATH="$tmpdir/boot/dtb" dtbs_install
 	fi
 fi
 
